packages/cli-kit/src/private/node/ui/components/Table/Table.tsx

```diff
@@ -0,0 +1,55 @@
+import ScalarDict from './ScalarDict.js'
+import {Row} from './Row.js'
+import React from 'react'
+import {Box} from 'ink'
+
+export interface TableProps<T extends ScalarDict> {
+  rows: T[]
+  columns: {[column in keyof T]: {header?: string; color?: string}}
+}
+
+export default function Table<T extends ScalarDict>({rows, columns: columnsConfiguration}: TableProps<T>) {
+  const columns = Object.entries(columnsConfiguration).map(([key, {header, color}]) => {
+    const headerWidth = String(header || key).length
+    const columnWidths = rows.map((row) => {
+      const value = row[key]
+
+      if (value === undefined || value === null) {
+        return 0
+      }
+
+      return String(value).length
+    })
+
+    return {
+      name: key,
+      width: Math.max(...columnWidths, headerWidth),
+      color,
+    }
+  })
+  const headings = Object.entries(columnsConfiguration).reduce(
+    (headings, [column, {header}]) => ({...headings, [column]: header || column}),
+    {},
+  )
+
+  return (
+    <Box flexDirection="column">
+      <Row
+        rowKey="heading"
+        line=" "
+        columns={columns.map((column) => ({...column, color: undefined}))}
```

Comment:
I get that you want to make it explicit that these rows have no color, but perhaps we can save a `map` here by passing `columns` as they are. I think it should be self evident that by default rows have no color. What do you think?

Reply:
If I pass `columns` as they are, headings and separators would be colored too, even though they should keep their standard color:
![10-02-obz4w-fo51t](https://user-images.githubusercontent.com/12963280/211558712-46d167f2-a416-4d9e-b3ba-c5d0f9fce0f2.jpg)

With the changes in https://github.com/Shopify/cli/pull/1040#discussion_r1064741876, I decided to introduce the flag `ignoreColumnColor`, instead of `.map`ping twice (for headings and separators)